### ISSUE TITLE
App crashes after every buy

### ISSUE THREAD

**Issue by fatings** [C1rdec/Poe-Lurker#360]
Steps:
1. Send an outgoing trade
2. Accept party invitation
3. Click the name of the user to join the hideout and activate the trade
4. Accept the trade
5. Send "thank you" message to the user
6. PoE Lurker UI disappears

Stack trace:
`[2022/11/06 06:59:30.621] Error [1] Object reference not set to an instance of an object.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Lurker.Patreon.Services.DatabaseService.<InsertAsync>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Lurker.UI.ViewModels.OutgoingbarViewModel.<InsertEvent>d__34.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Lurker.UI.ViewModels.OutgoingbarViewModel.<Lurker_TradeAccepted>d__33.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run(Window window)
   at Lurker.UI.App.Main() (Lurker.UI.AppBootstrapper)
`

Logs:
[2022-11-05.log](https://github.com/C1rdec/Poe-Lurker/files/9945305/2022-11-05.log)
[2022-11-06.log](https://github.com/C1rdec/Poe-Lurker/files/9945306/2022-11-06.log)

Notes:

- Used during 3.19 league without any problem, only started with Mayhem
- Using Centered UI mode (tried disabling for one trade and the result was the same)

**Comment by C1rdec** [comment]
Thanks I'll take a look. Maybe something changed with mayhem

**Comment by C1rdec** [comment]
I'm not sure what's going on but everything seems fine on my side.

Can you try to delete this file and try it again:

`C:\Users\\[user]\AppData\Roaming\PoeLurker\trade.db`

If this dont fix your issue let's connect on Discord to fix the issue

**Comment by ValidTactic** [comment]
I and another friend have this issue, as well. It has happened on every purchase I've made in event league Mayhem NRE001. I just deleted **trades.db** and made another purchase, but it still crashed. I then "deleted" the folder (renamed the folder to PoeLurker2), installed it fresh from GitHub, bought something, and it crashed again. It thanks them, self-kicks, and crashes before I can click my Hideout button. I just joined the Discord and will say something there.
